tracer/src/Datadog.Trace/Iast/Propagation/DefaultInterpolatedStringHandlerModuleImpl.cs

```diff
@@ -51,7 +55,16 @@ public static unsafe void FullTaintIfAnyTainted(IntPtr target, string? input)
             var rangesResult = new[] { new Range(0, 0, tainted!.Ranges[0].Source, tainted.Ranges[0].SecureMarks) };
             if (!targetIsTainted)
             {
-                taintedObjects.Taint(target, rangesResult);
+                // Safe guard to avoid memory leak
+                if (_taintedRefStructs.Count > 16)
```

Comment:
I think the capacity of the `_taintedRefStructs` object should be set such that we don't have to re-allocate. i.e. this value should be _less_ than the capacity. Otherwise setting the capacity doesn't achieve much

Reply:
Fixed